src/main/java/org/jabref/logic/l10n/Localization.java

```diff
@@ -48,53 +95,108 @@ public static void setLanguage(String language) {
         }
     }
 
+    /**
+     * Public access to the messages bundle for classes like AbstractView.
+     * @return The internally cashed bundle.
+     */
+    public static LocalizationBundle getMessages() {
+        // avoid situations where this function is called before any language was set
+        if (locale == null) {
+            setLanguage("en");
+        }
+        return localizedMessages;
+    }
+
+    /**
+     * Creates and caches the language bundles used in JabRef for a particular language. This function first loads
+     * correct version of the "escaped" bundles that are given in {@link l10n}. After that, it stores the unescaped
+     * version in a cached {@link LocalizationBundle} for fast access.
+     * @param locale Localization to use.
+     */
     private static void createResourceBundles(Locale locale) {
-        messages = ResourceBundle.getBundle(RESOURCE_PREFIX, locale, new EncodingControl(StandardCharsets.UTF_8));
-        menuTitles = ResourceBundle.getBundle(MENU_RESOURCE_PREFIX, locale, new EncodingControl(StandardCharsets.UTF_8));
+        ResourceBundle messages = ResourceBundle.getBundle(RESOURCE_PREFIX, locale, new EncodingControl(StandardCharsets.UTF_8));
+        ResourceBundle menuTitles = ResourceBundle.getBundle(MENU_RESOURCE_PREFIX, locale, new EncodingControl(StandardCharsets.UTF_8));
+        // Just for the case something really stupid happened.
+        if (messages == null || menuTitles == null) {
+            LOGGER.error("Could not load language translation files in " + Localization.class);
+            throw new NullPointerException();
+        }
+        localizedMessages = new LocalizationBundle(createLookupTable(messages));
+        localizedMenuTitles = new LocalizationBundle(createLookupTable(menuTitles));
     }
 
     /**
-     * In the translation, %0, ..., %9 is replaced by the respective params given
+     * Helper function to create a hash-table from the key/value pairs of a bundle.
+     * @param baseBundle JabRef language bundle with keys and values for translations.
+     * @return Lookup table for the baseBundle.
+     */
+    private static Hashtable<String, String> createLookupTable(ResourceBundle baseBundle){
+        final ArrayList<String> baseKeys = Collections.list(baseBundle.getKeys());
+        Hashtable<String, String> lookup = new Hashtable<>(baseKeys.size());
```

Comment:
HashTable is deprecated, better use HashMap.
And with java 8 streams and the Collection methods you an even make this code a lot nicer:
https://www.mkyong.com/java8/java-8-convert-list-to-map/